### DISCUSSION TITLE
419 page expired after upgrading to backpack 6 (and Laravel 10)

### DISCUSSION THREAD

**Discussion by de-raaf-media** [Laravel-Backpack/community-forum#576] [Q&A (Help)] [answered]
Hi,

I just finished all the upgrade steps to upgrade to BP 6 (from 5). I now get a message 419 page expired when I edit a crud. I know this has to do with CSRF tokens but I cannot explain why this is the case.

Did I miss a step?
Mark

**Comment by pxpm** [comment]
Hey @de-raaf-media 

I've experienced this myself and was due to permissions in the storage folder. 

I experienced this with Homestead, is that what you are using ? 

I tried it on windows, and had to make some changes in the vagrant file to assign the www-user as the owner of the project, has "vagrant" was taking over it. 

Also I was trying to use a custom folder: `c:\work`, that don't have full read/write permissions (on windows) and had to change the project to `c:\Users\MYUSERXXX\work`. 

If not you can also try a `php artisan cache:clear` and `php artisan basset:clear` (also make sure you are using `BASSET_DEV_MODE=true` in development to exclude any asset bugs)

Let me know how it goes.

**Comment by de-raaf-media** [comment]
Hi Pedro, I cleared the cache in the last step of the upgrade.

php artisan config:clear
php artisan cache:clear
php artisan view:clear

I now also cleared basset. There is now a basset folder in my storage so that seems to work fine. I am using Laravel Valet by the way.

It seems to be related to XHR routes like admin/page/search. 

Mark

**Reply by pxpm** [reply]
It happened to me on the pre-flight, but like I said, I think the reason was that the sessions were not properly persisted due to folder permissions. 

Can you check your browser console and network tab for errors? 

Also do your `APP_URL` in .env is set correctly as the url your use in your browser ? 

Cheers

**Reply by de-raaf-media** [reply]
I just updated my home office version as well. There it seems to work fine. So I'll have to check the office .env version. To be continued ;-)

**Comment by tabacitu** [comment]
Maybe it's a session issue? Do you have the same problem in an incognito window @de-raaf-media ? If so, a force logout would probably fix it (manually access the `/admin/logout` url).

**Reply by de-raaf-media** [reply]
That did not fix it but on another PC it is working fine. Maybe an env file issue

**Comment by tabacitu** [comment, marked as answer]
Are you sure your `APP_URL` is set correctly in your `.ENV` file?

**Reply by de-raaf-media** [reply]
Hi @tabacitu, @pxpm, this was indeed an .env file issue. I had no https in the URL for some reason. In my test application I do have https. In BP5 this was not an issue but in BP6 it is apparently. Thanks for the help guys.

**Reply by Fab-TOK** [reply]
Thanks you too much. It worked for me too.
I had the same problem and it blocked me for days. 
All you have to do is change the "**APP_URL**" variable in the .env file of the Laravel project. 
Mine used to be "APP_URL=localhost", but that didn't work.
Then I changed it to "APP_URL=http: //movie.test" and it worked. 
knowing that "http: //movie.test" is the **local** domain name I created for my project.
So all you have to do is change the "APP_URL" variable to the correct address. :-)

**Reply by tabacitu** [reply]
Thanks for the feedback everyone! Good news: we are adding a "check" after the installation, so new people don't run into this problem. Actually they might still run into this problem, but they'll get suggestions on how to fix it _right away_ if their env/config is misconfigured for some reason - https://github.com/Laravel-Backpack/basset/pull/72

**Comment by Fab-TOK** [comment]
Tell me what is the value of APP_URL now in the file .env ?

Le jeu. 20 juil. 2023 à 21:31, Joaquín Borges ***@***.***> a
écrit :

**Reply by JoacinB** [reply]
APP_URL=127.0.0.1

**Comment by Fab-TOK** [comment]
try to change it to
APP_URL=127.0.0.1/8000
save the file and refresh the web page


Le ven. 21 juil. 2023 à 02:08, Joaquín Borges ***@***.***> a
écrit :

**Comment by Fab-TOK** [comment]
ooh my bad, instead APP_URL=127.0.0.1:8000

not /8000

**Comment by JoacinB** [comment]
Tried both but no success

**Reply by JoacinB** [reply]
This url works and shows the user 1 data: http://127.0.0.1:8000/admin/user/1/show
But this one returns error 419: http://127.0.0.1:8000/admin/user

**Comment by JoacinB** [comment]
I found the solution guys!!!
 
There is a file named datatables_logic.blade.php which has configurations for an AJAX request at line 266:


![image](https://github.com/Laravel-Backpack/community-forum/assets/73504597/15cd203c-82e0-4cbe-9c11-cc0b00b1bc56)

I added:
```
              "headers": {
                'X-CSRF-TOKEN': '<?php echo e(csrf_token()); ?>'
              },
```
and it worked!!!

**Reply by mollevi** [reply]
worked.

**Comment by sadorect** [comment]
I have implemented all the suggestions here and there has been no change still.
I'm using 
Laravel 10.15
Backpack 6
APP_URL is correct
csrf_token is included on page when you check the page source.

On the list page, the 419 error only shows on a frame which disappears after you click outside but, then, the Data is not listed which is the actual concern.

I have cleared cache all the time. What more is there to do please?

**Reply by pxpm** [reply]
@sadorect please check in developer console if `common.js` is beeing loaded correctly.

**Reply by sadorect** [reply]
Thank you @pxpm 
Amazingly, I just found it resolved after I hibernated the PC and came back to it. Not that the hibernation was deliberately targetted at the problem, I only needed to get away from the system and when I came back, we're in business.

Thanks a lot!

**Comment by colincameron** [comment]
I'm getting the same error on a clean installation of Backpack 6 on Laravel 10. My issue was with the new Basset library and a local Docker environment. I had to add the `php artisan storage:link` command to my Dockerfile for the asset directory to be created.
